eva-accession-clustering-automation/create_clustering_properties.py

```diff
@@ -0,0 +1,186 @@
+        add_spring_properties(property_line)
+    return path
+
+
+def get_properties(profile, private_config_file, private_config_xml_file):
+    if private_config_xml_file:
+        return get_properties_from_xml_file(profile, private_config_xml_file)
+
+    if private_config_file:
+        private_config_args = get_args_from_private_config_file(private_config_file)
+        if private_config_args.get('github_token'):
+            settings = get_eva_settings_xml_string(private_config_args['github_token'])
+            return get_properties_from_xml_string(profile, settings)
+
+    logger.error('Must provide either the private config xml path using parameter --private-config-xml-file or the '
+                 'github token by including the "github_token" field in the config file --private-config-file')
+    sys.exit(1)
+
+
+def get_properties_path(source, vcf_file, project_accession, assembly_accession, output_directory):
+    path = output_directory + '/' + assembly_accession
+    if source.upper() == 'VCF':
+        path += '_' + os.path.basename(vcf_file) + '_' + project_accession
+    path += '.properties'
+    return path
+
+
+def preliminary_check(source, vcf_file, project_accession):
+    """
+    This checks must pass in order to run the script
+    """
+    check_valid_sources(source)
+    check_vcf_source_requirements(source, vcf_file, project_accession)
+
+
+def add_clustering_properties(property_line, assembly_accession, project_accession, source, vcf_file):
+    property_line.write('spring.batch.job.names=' + get_job_name(source) + '\n')
+    property_line.write('\n')
+    property_line.write('parameters.assemblyAccession=' + assembly_accession + '\n')
+    if vcf_file: property_line.write('parameters.vcf=' + vcf_file + '\n')
+    if project_accession: property_line.write('parameters.projectAccession=' + project_accession + '\n')
+
+
+def get_job_name(source):
+    if source.upper() == 'MONGO':
+        return 'CLUSTERING_FROM_MONGO_JOB'
+    elif source.upper() == 'VCF':
+        return 'CLUSTERING_FROM_VCF_JOB'
+
+
+def add_accessioning_properties(property_line):
+    property_line.write('\n')
+    property_line.write('parameters.chunkSize=100' + '\n')
+    property_line.write('\n')
+    property_line.write('accessioning.instanceId=instance-01' + '\n')
+    property_line.write('accessioning.submitted.categoryId=ss' + '\n')
+    property_line.write('accessioning.clustered.categoryId=rs' + '\n')
+    property_line.write('\n')
+    property_line.write('accessioning.monotonic.ss.blockSize=100000' + '\n')
+    property_line.write('accessioning.monotonic.ss.blockStartValue=5000000000' + '\n')
+    property_line.write('accessioning.monotonic.ss.nextBlockInterval=1000000000' + '\n')
+    property_line.write('accessioning.monotonic.rs.blockSize=100000' + '\n')
+    property_line.write('accessioning.monotonic.rs.blockStartValue=3000000000' + '\n')
+    property_line.write('accessioning.monotonic.rs.nextBlockInterval=1000000000' + '\n')
```

Comment:
why not a single multiline string? to keep indentation?
```
property_line.write("""
parameters.chunkSize=100

accessioning.instanceId=instance-01
...
""")
```

Reply:
I tried your suggestion and liked it better!